src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    let newSeparator = separator;
+
+    if (newSeparator === null) {
+      newSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        if (this[i] === undefined || this[i] === null) {
+          str += '';
+        } else {
+          str += this[i];
+        }
+      } else {
+        if (this[i] === undefined || this[i] === null) {
+          str += '' + newSeparator;
+        } else {
```

Comment:
You need a separate condition, that will add an element or an empty string. It can be done even in a single row using [Nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing).